### DISCUSSION TITLE
Shortcut to go to a specific Note / PDF / Collection or open any link

### DISCUSSION THREAD

**Discussion by mjthoraval** [windingwind/zotero-actions-tags#398] [Q&A]
As I am working on a project, I would like to be able to go quickly to a specific location.
For example, to go quickly to the main note I am working on for the project, or the book I am reading, or even a link to an external application like Evernote.
The copy link action of this plugin can already target any destination in Zotero, so the shortcut would just be to open that link.

**Comment by windingwind** [comment]
Do you mean you already have a link in the clipboard and want to open that link?

**Reply by mjthoraval** [reply]
I want the shortcut to always open the same link.
So the link will be written inside the script of the action.

**Reply by windingwind** [reply]
```js
const ZoteroPane = Zotero.getActiveZoteroPane();
const uri = `zotero://select`;
ZoteroPane.loadURI(uri);
```

**Reply by mjthoraval** [reply]
Thank you.
I am using this script to link to a note with the Better Notes plugin:
```JavaScript
const ZoteroPane = Zotero.getActiveZoteroPane();
const uri = `zotero://note/u/WPLQPABX/`;
ZoteroPane.loadURI(uri);
```
with the shortcut `Ctrl`+`H`.

If no item is selected in My Library, it opens the note correctly.
If one item is already selected in My Library, it opens the note twice.
If 2 items are selected, it opens the note 3 times.
...

Can this be fixed?

Also, what should I modify if I want the note to open in a separate window or as a tab, independently of the BN option?

**Reply by ramonmi** [reply]
You can use this to avoid repeated openings.
```
if (item) {
	return;
}
const ZoteroPane = Zotero.getActiveZoteroPane();
const uri = `zotero://note/u/WPLQPABX/`;
ZoteroPane.loadURI(uri);
```

>Also, what should I modify if I want the note to open in a separate window or as a tab, independently of the BN option?

You can get the note ID and open the note using it. Three ways to open the note are supported: in a separate window, as a BN tab, or as a BN window.

```
// use the below four line to get the noteID and remember it
if (!item) { return; }
if (item.isNote()) {
	return `${item.id}`;
}
```

```
// user below script to open the note
if (!Zotero.BetterNotes?.api.note?.insert) {
	return "[Custom-Template] Better Notes for Zotero > 1.1.4-21 is not installed or disabled.";
}

if (item) {
	return;
}
const noteID="9093"; // change it
Zotero.getActiveZoteroPane().openNoteWindow(noteID); // open in default separate window
// Zotero.BetterNotes.hooks.onOpenNote(noteID, "window"); // open in Better Notes window
// Zotero.BetterNotes.hooks.onOpenNote(noteID, "tab"); // open in Better Notes tab
```
